tensorflow/core/common_runtime/dml/dml_operator_helper.h

```diff
@@ -144,6 +144,22 @@ class BatchNormGradShapeHelper : public ShapeHelper {
       const InitializationHelper* initialization_helper) const override;
 };
 
+class SoftmaxXentWithLogitsShapeHelper : public ShapeHelper {
```

Comment:
Don't add new shape helpers in this file. Only shape helpers that can be used by many kernels should be here. The BatchNorm helpers are relics from the past that should be moved out eventually.